libraries/AP_GPS/AP_GPS_UBLOX.cpp

```diff
@@ -979,16 +979,22 @@ AP_GPS_UBLOX::_parse_gps(void)
                         gnssCount++;
                     }
                 }
-
                 for(int i = 0; i < _buffer.gnss.numConfigBlocks; i++) {
-                    // Reserve an equal portion of channels for all enabled systems
+                    // Reserve an equal portion of channels for all enabled systems that supports it
                     if(gps._gnss_mode[state.instance] & (1 << _buffer.gnss.configBlock[i].gnssId)) {
-                        if(GNSS_SBAS !=_buffer.gnss.configBlock[i].gnssId) {
+                        if(GNSS_SBAS !=_buffer.gnss.configBlock[i].gnssId && (_hardware_generation > UBLOX_M8 || GNSS_GALILEO !=_buffer.gnss.configBlock[i].gnssId)) {
```

Comment:
This will still modify behaviour on non-M8

Reply:
This is as intended. 
Hardware > M8 will keep the previous behaviour, aka allocate the channels as it was in the past. Basically everything newer than M8 will go into the else only for configuring SBAS. Am I reading it wrong?